### ISSUE TITLE
Build Breakage: ld: library not found for -lmuparser

### ISSUE THREAD

**Issue by dsMrP** [fish-shell/fish-shell#4379]
Just pulled the latest sources and attempted to build the following hash on OSX:

    [david@fumes 11:24:03]:~/d/fish-shell (master=) git log --oneline -1
    76609de4 (HEAD -> master, origin/master, origin/HEAD) pacman-ish completions: Complete files for `-Qo` and `-Qp`
    [david@fumes 11:24:17]:~/d/fish-shell (master=)

with:

    [david@fumes 11:20:54]:~/d/fish-shell (master=) xcodebuild install 
    === BUILD AGGREGATE TARGET generate-version-header OF PROJECT fish WITH THE DEFAULT CONFIGURATION (Release) ===
    
    ...
    
    Ld /tmp/fish.dst/usr/local/bin/fish normal x86_64
        cd /Users/david/dev/fish-shell
        export MACOSX_DEPLOYMENT_TARGET=10.7
        /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk -L/Users/david/dev/fish-shell/build/Release -F/Users/david/dev/fish-shell/build/Release -filelist /Users/david/dev/fish-shell/build/fish.build/Release/fish_shell.build/Objects-normal/x86_64/fish.LinkFileList -mmacosx-version-min=10.7 -dead_strip -Xlinker -object_path_lto -Xlinker /Users/david/dev/fish-shell/build/fish.build/Release/fish_shell.build/Objects-normal/x86_64/fish_lto.o -Xlinker -final_output -Xlinker /usr/local/bin/fish -stdlib=libc++ -flto=thin -Xlinker -cache_path_lto -Xlinker /Users/david/dev/fish-shell/build/LTOCache -lncurses -lpcre2 -lmuparser -Xlinker -dependency_info -Xlinker /Users/david/dev/fish-shell/build/fish.build/Release/fish_shell.build/Objects-normal/x86_64/fish_dependency_info.dat -o /tmp/fish.dst/usr/local/bin/fish
    ld: library not found for -lmuparser
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    
    ** INSTALL FAILED **
    
    
    The following build commands failed:
    	Ld /tmp/fish.dst/usr/local/bin/fish normal x86_64
    (1 failure)
    [david@fumes 11:21:10]:~/d/fish-shell (master=)

**Comment by dsMrP** [comment]
Added a patch file to fix the project build, however still not getting the command line build to work at this stage...

[project.txt](https://github.com/fish-shell/fish-shell/files/1278537/project.txt)

**Comment by krader1961** [comment]
I have a different xcode project definition in my private fish fork that works for me. If you install the attached file as *fish.xcodeproj/project.pbxproj* (drop the *.txt* suffix) you should be able to build fish. FWIW, I don't normally use Xcode to work on or build fish. Instead I use the portable `./configure --prefix=/usr/local; make test; make install` CLI sequence. I suspect that's true for everyone else because it's not unusual to see the Xcode project definition get out of sync with other changes to the fish code base.

[project.pbxproj.txt](https://github.com/fish-shell/fish-shell/files/1278519/project.pbxproj.txt)

**Comment by dsMrP** [comment]
Thanks the project change did indeed resolve the issue. I build fish based on the recommendation at the bottom of the git hub page for Xcode Build and Install. If this is not the correct way to build fish, then perhaps that information should be updated?

**Comment by faho** [comment]
@dsMrP: I'm assuming you've updated from a version without muparser to one with it.

Have you tried cleaning the build artifacts? For me, I needed to do a `git clean -xdf` (`make clean` wasn't enough). Unfortunately I have no idea how xcode works.

**Comment by zanchey** [comment]
No, the problem occurs with a clean checkout.

**Comment by dsMrP** [comment]
The issue is the target muparser target isn't referenced correctly in the base build dependencies.

**Comment by krader1961** [comment]
FWIW, I'm reasonably confident my *project.pbxproj* file that I attached to this issue above is correct. I can use it to both build fish with Xcode and run `make lint-all`. I spent several hours figuring out how to update the Xcode project definition when it wasn't clear the more experienced Xcode developers on the project would do so given the recent controversy about including a dependency on the MuParser library. I make no claim that my Xcode project definition is optimal. But it is clearly better than the definition currently included in the project in as much as "it works" whether you're building with Xcode or using its artifacts via `make lint-all`.
